modules/virt-creating-a-service-from-a-virtual-machine.adoc

```diff
@@ -0,0 +1,177 @@
+====
+Labels on a virtual machine are passed through to the pod. The labels on
+the `VirtualMachine`, for example `special: key`, must match the labels in
+the Service  YAML `selector` attribute, which you create later
+in this procedure.
+====
+
+. Save the virtual machine YAML to apply your changes.
+
+. Edit the service YAML to configure the settings necessary to create and expose the Service object.
++
+
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: vmservice <1>
+  namespace: example-namespace <2>
+spec:
+  ports:
+  - port: 27017
+    protocol: TCP
+    targetPort: 22 <3>
+  selector:
+    special: key <4>
+  type: ClusterIP <5>
+----
+<1> Specify the `name` for the service you are creating and exposing.
+<2> Specify `namespace` in the `metadata` section of the Service YAML that corresponds to the `namespace` you specify in the virtual machine YAML.
+<3> Add `targetPort: 22`, exposing the service on SSH port `22`.
+<4> In the `spec` section of the Service YAML, add `special: key` to the `selector` attribute, which corresponds to the `labels` you added in the virtual machine YAML configuration file.
+<5> In the `spec` section of the Service YAML, add `type: ClusterIP` for a
+ClusterIP service. To create and expose other types of services externally, outside of the cluster, such as `NodePort` and `LoadBalancer`, replace
+`type: ClusterIP` with `type: NodePort` or `type: LoadBalancer`, as appropriate.
++
+
+. Save the service YAML to apply your changes.
+. Create the `ClusterIP` service.
++
+
+[source,terminal]
+----
+$ oc create -f <service_name>.yaml
+----
+
++
+. Start the virtual machine. If the virtual machine is already running, restart it.
++
+
++
+. Query the Service object to verify it is available and is configured with type `ClusterIP`.
++
+
+.Verification steps
+* Run the `oc get service command`.
++
+[source, terminal]
+----
+$ oc get service
+----
++
+.Example output
+[source, terminal]
+----
+NAME        TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)     AGE
+vmservice   ClusterIP   172.30.3.149   <none>        27017/TCP   2m
+----
++
+** As shown from the `AGE`, `vmservice` has been running for 2 minutes.
+** The `TYPE` displays as `ClusterIP`, as you specified in the Service YAML.
+
+. Connect to the virtual machine using the service IP and service port.
+.. List all information
+about a project’s pod deployment by running this command and providing the node name:
++
+
+[source,terminal]
+----
+$ oc get node <node_name> -o wide
+----
++
+
+.Example output
+[source,terminal]
+----
+NAME                           STATUS    ROLES     AGE       VERSION           INTERNAL-IP      OS-IMAGE                            KERNEL-VERSION          CONTAINER-RUNTIME
+ip-172-18-0-199.ec2.internal   Ready     infra     1d        v1.10.0+b81c8f8   54.172.185.130   Fedora cloud container disk demo    3.10.0-862.el7.x86_64   docker://1.13.1
+----
++
+
+.. Connect to the virtual machine. In the `ssh` command, replace
```

Comment:
Connect from where?
Connecting to an exposed pod (or VM, in our case) via a ClusterIP must be done from an object inside the cluster - a pod or another VM.
The simplest way to do it, IMO, is creating another VM. You can use the example of the VM above to create another VM, just change the following fields in the new YAML:
 a. Give a different `name`.
 b. Remove the `special: key` label (you can remove the entire `labels` entry, as this was the only label there.
Then connect to the new VM with `virtctl -n example-namespace console <new-vm-name>`, and run
`ssh fedora@172.30.3.149 -p 27017`
where `172.30.3.149` is the `CLUSTER-IP` of the **service**, `fedora` is the username of the VM.
Getting the command prompt of the **original** VM is the verification.

Reply:
Created more steps for this. Please verify that it is correct. Thanks.